sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -288,7 +287,7 @@ object RowEncoder {
 
       StaticInvoke(
         ArrayBasedMapData.getClass,
-        ObjectType(classOf[Map[_, _]]),
+        ObjectType(classOf[scala.collection.immutable.Map[_, _]]),
```

Comment:
@srowen `scala.collection.Map` is imported at the beginning of `RowEncoder` and `ScalaReflection`.